mhs/spineroutelookup/lookup/tests/test_redis_cache.py

```diff
@@ -0,0 +1,111 @@
+import unittest.mock
+
+import redis
+from utilities.test_utilities import async_test
+
+from lookup import redis_cache
+
+REDIS_HOST = "host"
+REDIS_PORT = 1234
+USE_TLS = False
+
+ODS_CODE = "ods"
+INTERACTION_ID = "interaction-id"
+CACHE_KEY = ODS_CODE + "-" + INTERACTION_ID
+
+VALUE_DICTIONARY = {
+    "key1": "value1",
+    "key2": ["list_entry_1", "list_entry_2"],
+    "key3": {
+        "nested_key_1": "nested_value_1",
+        "nested_key_2": "nested_value_2"
+    }
+}
+
+VALUE_DICTIONARY_JSON = '{"key1": "value1", "key2": ["list_entry_1", "list_entry_2"], "key3": {' \
+                        '"nested_key_1": "nested_value_1", "nested_key_2": "nested_value_2"}}'
+
+FIFTEEN_MINUTES_IN_SECONDS = 900
+
+
+class TestRedisCache(unittest.TestCase):
+
+    def setUp(self) -> None:
+        # Mock the redis.Redis() constructor
+        patcher = unittest.mock.patch.object(redis, "Redis")
+        self.mock_redis_constructor = patcher.start()
+        self.addCleanup(patcher.stop)
+
+        # Mock the Redis client class itself
+        self.mock_redis = unittest.mock.MagicMock()
+        self.mock_redis_constructor.return_value = self.mock_redis
+
+    def test_redis_params_are_passed(self):
+        redis_cache.RedisCache(REDIS_HOST, REDIS_PORT, use_tls=USE_TLS)
+
+        self.mock_redis_constructor.assert_called_with(host=REDIS_HOST, port=REDIS_PORT, ssl=USE_TLS)
+
+    def test_tls_is_enabled_by_default(self):
+        redis_cache.RedisCache(REDIS_HOST, REDIS_PORT)
+
+        self.mock_redis_constructor.assert_called_with(host=REDIS_HOST, port=REDIS_PORT, ssl=True)
+
+    @async_test
+    async def test_should_retrieve_value_from_store_if_exists(self):
+        cache = redis_cache.RedisCache(REDIS_HOST, REDIS_PORT)
+        self.mock_redis.get.return_value = VALUE_DICTIONARY_JSON.encode()
+
+        value = await cache.retrieve_mhs_attributes_value(ODS_CODE, INTERACTION_ID)
+
+        self.assertEqual(value, VALUE_DICTIONARY)
+        self.mock_redis.get.assert_called_with(CACHE_KEY)
```

Comment:
i think this is a bit of an explicit assertion, this may make the test more brittle than it needs to be. The fact we get the correct value back shows the behaviour is working correctly IMO.

Reply:
As discussed offline, the goal here was to confirm that the correct argument was passed to the `mock_redis.get` call, rather than asserting that this specific method was called. Unfortunately, as far as I can tell this needs to be done using an assert statement, rather than when setting up the mock.